plugin/cache/item.go

```diff
@@ -28,9 +28,16 @@ func newItem(m *dns.Msg, now time.Time, d time.Duration) *item {
 	i.Authoritative = m.Authoritative
 	i.AuthenticatedData = m.AuthenticatedData
 	i.RecursionAvailable = m.RecursionAvailable
-	i.Answer = m.Answer
-	i.Ns = m.Ns
+	i.Answer = make([]dns.RR, len(m.Answer))
+	i.Ns = make([]dns.RR, len(m.Ns))
 	i.Extra = make([]dns.RR, len(m.Extra))
+
+	for j, r := range m.Answer {
+		i.Answer[j] = dns.Copy(r)
```

Comment:
do we really need `Copy` or can we just use `copy`?

Reply:
For the bug described here `copy` is fine, I was trying to do a deep copy to prevent the same issue from happening one level deeper. I can't think of any reason to modify the response struct itself. I defer to you - do you prefer that I simply shallow copy?